src/core/qgsmaplayer.cpp

```diff
@@ -3253,7 +3253,24 @@ QString QgsMapLayer::generalHtmlMetadata() const
         continue;
 
       const QVariant propValue = customProperty( key );
-      metadata += QStringLiteral( "<tr><td class=\"highlight\">%1</td><td>%2</td></tr>" ).arg( key.toHtmlEscaped(), propValue.toString().toHtmlEscaped() );
+      QString stringValue = QString();
+      if ( propValue.canConvert<QStringList>() && propValue.toStringList().count() > 1 )
```

Comment:
We shouldn't use canConvert anymore -- the meaning of this changed in Qt 6, to mean "is there anyway i can possibly create a string list from this object", and the answer will always be yes, because it CAN create an empty list :man_shrugging: 

Instead, you need to explicitly check the variant metatype to determine whether it contains a list value.